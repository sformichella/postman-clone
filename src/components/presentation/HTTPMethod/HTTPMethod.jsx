import React from "react";
import PropTypes from "prop-types";

export default function HTTPMethod({ method }) {
  return (
    <>
      <label htmlFor={method}>{method}</label>
      <input
        id={method}
        type="radio"
        name="method"
        value={method}
      />
    </>
  );
}

HTTPMethod.propTypes = {
  method: PropTypes.string.isRequired
};
