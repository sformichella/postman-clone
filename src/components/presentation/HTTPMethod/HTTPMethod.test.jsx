import React from "react";
import { render, cleanup } from "@testing-library/react";
import HTTPMethod from "./HTTPMethod";

describe("HTTPMethod component", () => {
  afterEach(() => cleanup());
  it("renders HTTPMethod", () => {
    const { asFragment } = render(<HTTPMethod />);
    expect(asFragment()).toMatchSnapshot();
  });
});
