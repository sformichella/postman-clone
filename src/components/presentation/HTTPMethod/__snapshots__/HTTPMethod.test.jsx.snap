// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTTPVerb component renders HTTPVerb 1`] = `
<DocumentFragment>
  <label />
  <input
    name="method"
    type="radio"
    value=""
  />
</DocumentFragment>
`;
